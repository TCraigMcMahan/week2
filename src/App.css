.App {
  text-align: center;
}

.App-header {
  background-color: white;
  height: 50px;
  padding: 10px;
  border-top: 0px;
  border-left: 0px;
  border-right: 0px;
  border-bottom: 2px;
  border-bottom-color: lightgray;
  border-style: solid;
  margin-left: 10px;
  margin-bottom: 25px;
  width: 1075px;
  max-width: 1100px;
}

.App-title {
  font-size: 1.5em;
  text-align: left;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.App-card-container {
  display: flex;
  flex-wrap: wrap;
  height: 600px;
  max-height: 600px;
  width: 1100px;
  max-width: 1100px;
  align-items: center;
  background-color: white;
}

.App-card-container::before {
  content: " ";
  position: absolute;
  z-index: -1;
  top: 50px;
  left: 8px;
  right: 20px;
  width: 95px;
  border-bottom: 1.5px solid black;
}

.App-missed-card-container {
  display: flex;
  flex-wrap: wrap;
  height: 350px;
  max-height: 350px;
  width: 1100px;
  max-width: 1100px;
  align-items: center;
  background-color: white;
}

.App-missed-card-container::before {
  content: " ";
  position: absolute;
  z-index: -1;
  top: 725px;
  left: 8px;
  right: 20px;
  width: 240px;
  border-bottom: 1.5px solid black;
}
  