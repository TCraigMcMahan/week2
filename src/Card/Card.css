* {
    box-sizing: border-box;
}

.side {
    flex: 30%;
    padding: 10px;
}

.main {
    display: flex;
    flex-wrap: nowrap;
    flex-direction: column;
    flex: 70%;
    padding: 10px;
}

.row {
    height: 275px;
    max-height: 275px;
    width: 525px;
    max-width: 525px;
    background-color: white;
    border: lightgray;
    border-width: 2px;
    border-style: solid;
    margin: 10px;
}

.Card-img {
    float: left;
    width: 250px;
    height: 250px;
    padding: 10px;
}

.Title {
    box-align: left;    
}

.Author-img {
    width: 50px;
    height: 50px;
    border-radius: 50%;
    padding: 3px;
    background-size: auto;
}

.authorInfo {
    display: flex;
    flex-wrap: nowrap;
    flex-direction: row;
    height: 75px;
    max-height: 75px;
    font-size: 12px;
}


.authorInfo-namedate {
    display: flex;
    flex-wrap: nowrap;
    flex-direction: column;
    padding: 10px;
    height: 75px;
    max-height: 75px;
    width: 200px;
    font-size: 15px;
}

.titleStyle {
    font-weight: bold;
    text-align: left;    
}

.bookmarkStyle {
    width: 50%;
    height: 50%;
}

.authorNameStyle {
    font-weight: bold;
    text-align: left;
}

.authorMemberContainer {
    height: 50px;
    width: 50px;
    margin-left: 5px;
    background-image: url('../img/arcs.png');
}

.upperScreen {
    display: flex;
    flex-wrap: nowrap;
    flex-direction: column;
    height: 150px;
    max-height: 150px;
    font-size: 15px;
}

.bookmarkContainer {
    height: 25px;
    width: 25px;  
    padding-top: 10px;
    margin-left: 10px;
}

.bookmarkContainer img {
    height: auto;
    width: 90%;
}