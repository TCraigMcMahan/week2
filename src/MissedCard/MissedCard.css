.Missed-row {
    height: 275px;
    max-height: 275px;
    width: 345px;
    max-width: 345px;
    background-color: white;
    border: lightgray;
    border-width: 2px;
    border-style: solid;
    margin: 10px;
}

.Missed-Card-img {
    float: left;
    width: 345px;
    height: 100px;
    padding: 10px;
    object-fit: cover;
}

.missedAuthorInfo {
    display: flex;
    flex-wrap: nowrap;
    padding: 10px;
    height: 150px;
    max-height: 150px;
}

.missedTitleStyle {
    text-align: left;
    padding-left: 25px;
    font-weight: bold;
}

.missedDescriptionStyle {
    text-align: left;
    padding-left: 25px;
}

.missedUpperScreen {
    display: flex;
    flex-wrap: nowrap;
    flex-direction: column;
    height: 200px;
    max-height: 200px;
    font-size: 15px;
}