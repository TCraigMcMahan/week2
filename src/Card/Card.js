import React, { Component } from 'react';
import './Card.css';
import Audio from '../Icons/Audio';
import '../Icons/Audio.css';
import speaker from '../img/icon-speaker.png';
import star from '../img/icon-star.svg';
import bookmark from '../img/icon-bookmark.svg';
import PropTypes from 'prop-types';

class Card extends Component {
    render() {

        const {
            title,
            image,
            description,
            link,
            author,
            postedDate,
            minutesToRead,
            hasAudioAvailable,
            memberPreview
        } = this.props.card;


        const showMember = (props) => {
            if (author.isMediumMember) {
                return (
                    <div className="authorMemberContainer">
                        <img className="Author-img" src={author.image} alt="" />
                    </div>
                )
            }
            else {
                return (
                    <div>
                        <img className="Author-img" src={author.image} alt="" />
                    </div>
                )
            }
        }

        const getDate = (props) => {
            const monthNum = postedDate.toString().substring(5, 7);
            let monthAbbrev = '';
            const dayNum = postedDate.toString().substring(8, 10);
            let modDayNum = dayNum;
            let dateStr = '';

            switch (parseInt(monthNum, 10)) {
                case 1:
                    monthAbbrev = 'Jan';
                    break;
                case 2:
                    monthAbbrev = 'Feb';
                    break;
                case 3:
                    monthAbbrev = 'Mar';
                    break;
                case 4:
                    monthAbbrev = 'Apr';
                    break;
                case 5:
                    monthAbbrev = 'May';
                    break;
                case 6:
                    monthAbbrev = 'Jun';
                    break;
                case 7:
                    monthAbbrev = 'Jul';
                    break;
                case 8:
                    monthAbbrev = 'Aug';
                    break;
                case 9:
                    monthAbbrev = 'Sep';
                    break;
                case 10:
                    monthAbbrev = 'Oct';
                    break;
                case 11:
                    monthAbbrev = 'Nov';
                    break;
                case 12:
                    monthAbbrev = 'Dec';
                    break;
                default:
                    monthAbbrev = '';
            }

            if (dayNum[0] === '0') {
                modDayNum = dayNum[1];
            }
            else {
                modDayNum = dayNum;
            }

            dateStr = monthAbbrev + ' ' + modDayNum;

            return dateStr;
        }

        return (

            <div className="row">
                <div className="side" >
                    <img className="Card-img" src={image} alt="" />
                </div>
                <div className="main">
                    <div className="upperScreen">
                        <Audio hasAudioAvailable={hasAudioAvailable} memberPreview={memberPreview} />
                        <div className="titleStyle"> {title} </div>
                        <br></br>
                        <div align="left">{description}</div>
                    </div>
                    <div className="authorInfo" >
                        <div>
                            {showMember(author.isMediumMember)}
                        </div>
                        <div className="authorInfo-namedate">
                            <div className="authorNameStyle">{author.name}</div>
                            <div align="left"> {getDate()} - {minutesToRead} min read</div>
                        </div>
                        <div className="bookmarkContainer">
                        <img className="bookmarkStyle" src={bookmark} alt="Bookmark" />
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

Card.propTypes = {
    title: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    description: PropTypes.string,
    author: PropTypes.instanceOf(Object),
    postedDate: PropTypes.string.isRequired,
    minutesToRead: PropTypes.number,
    hasAudioAvailable: PropTypes.bool.isRequired,
    memberPreview: PropTypes.bool.isRequired
};

Card.defaultProps = {
    name: 'Anonymous',
    memberPreview: false,
}

export default Card;
