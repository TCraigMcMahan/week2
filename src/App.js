import React, { Component } from 'react';
import './App.css';
import Card from './Card/Card.js';
import MissedCard from './MissedCard/MissedCard.js';

import articles from './your-articles.json';
import missedArticles from './missed-articles.json';

class App extends Component {
  render() {

    return (
      <div className="App">
        <div className="App-header">
          <div className="App-title">For you</div>
        </div >
        <div className="App-card-container" >
          {articles.map((card, idx) => {
            return <Card key={idx} card={card} />;
          })}
        </div >

        <div className="App-header">
          <div className="App-title">In case you missed it</div>
        </div >
        <div className="App-missed-card-container" >
          {missedArticles.map((card, idx) => {
            return <MissedCard key={idx} missedcard={card} />;
          })}
        </div >
      </div>
    );
  }
}

export default App;
