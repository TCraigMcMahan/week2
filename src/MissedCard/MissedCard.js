import React, { Component } from 'react';
import './MissedCard.css';
import '../Card/Card.css';
import Audio from '../Icons/Audio.js';
import '../Icons/Audio.css';
import speaker from '../img/icon-speaker.png';
import star from '../img/icon-star.svg';
import bookmark from '../img/icon-bookmark.svg';
import PropTypes from 'prop-types';

class MissedCard extends Component {
    
    render() { 
        const {
            title,
            image,
            description,
            link,
            author,
            postedDate,
            minutesToRead,
            hasAudioAvailable,
            memberPreview
        } = this.props.missedcard;

        const showMember = (props) => {
            if ( author.isMediumMember ) {
                return (
                    <div className="authorMemberContainer">
                        <img  className="Author-img" src={author.image} alt="" />
                    </div>
                )                   
            }
            else {
                return (
                    <div>
                        <img  className="Author-img" src={author.image} alt="" />
                    </div>
                )
            }
        }

        const getDate = (props) => {
            const monthNum = postedDate.toString().substring(5,7);
            let monthAbbrev = '';
            const dayNum = postedDate.toString().substring(8,10);
            let modDayNum = dayNum;
            let dateStr = '';

            switch (parseInt(monthNum, 10)) {
                case 1:
                    monthAbbrev = 'Jan';
                    break;
                case 2:
                    monthAbbrev = 'Feb';
                    break;
                case 3:
                    monthAbbrev = 'Mar';
                    break;
                case 4:
                    monthAbbrev = 'Apr';
                    break;
                case 5:
                    monthAbbrev = 'May';
                    break;
                case 6:
                    monthAbbrev = 'Jun';
                    break;
                case 7:
                    monthAbbrev = 'Jul';
                    break;
                case 8:
                    monthAbbrev = 'Aug';
                    break;    
                case 9:
                    monthAbbrev = 'Sep';
                    break;
                case 10:
                    monthAbbrev = 'Oct';
                    break;
                case 11:
                    monthAbbrev = 'Nov';
                    break;
                case 12:
                    monthAbbrev = 'Dec';
                    break;
                default:
                    monthAbbrev = '';    
            }
            
            if (dayNum[0] === '0') {
                modDayNum = dayNum[1];
            }
            else {
                modDayNum = dayNum;
            }

            dateStr = monthAbbrev + ' ' + modDayNum;

            return dateStr;
        }

        return (
            <div className="Missed-row">
                <div className="missedUpperScreen">
                <img className="Missed-Card-img" src={image} alt="" />
                <Audio hasAudioAvailable={hasAudioAvailable} memberPreview={memberPreview} />
                <div className="missedTitleStyle"> {title} </div>
                <br></br>
                <div  className="missedDescriptionStyle">{description}</div>
                </div>
                <div className="missedAuthorInfo" >
                    {showMember(author.isMediumMember)}         
                    <div className="authorInfo-namedate">
                        <div className="authorNameStyle">{author.name}</div>
                        <div align="left"> {getDate()} - {minutesToRead} min read</div>
                    </div>
                    <div className="bookmarkContainer">
                    <img className="bookmarkStyle" src={bookmark} alt="Bookmark" />
                    </div>
                </div>
            </div>
        );
    }
}

MissedCard.propTypes = {
    title: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    description: PropTypes.string,
    author: PropTypes.instanceOf(Object),
    postedDate: PropTypes.string.isRequired,
    minutesToRead: PropTypes.number,
    hasAudioAvailable: PropTypes.bool.isRequired,
    memberPreview: PropTypes.bool.isRequired
};

MissedCard.defaultProps = {
    name: 'Anonymous',
    memberPreview: false,
}

export default MissedCard;