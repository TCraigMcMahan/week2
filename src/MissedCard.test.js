import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import './App.css';
import MissedCard from './MissedCard/MissedCard';
import './MissedCard/MissedCard.css';
import './Card/Card.css';
import testCard from './cardTest.json';

describe('MissedCard', () => {

    it('renders without crashing with data', () => {
/*         const minutes = 3;
        const boolean = false;
        const author = {
            name: "Kris Gage",
            image: "https://cdn-images-1.medium.com/fit/c/80/80/1*e7o4kbk8ofT6eKaLrO0jaw.png",
            isMediumMember: false
        };
 */
        const div = document.createElement('div');
        const element = testCard.map((card, index) => {
            return <MissedCard key={index} missedcard={card} />
        });
        /*         ReactDOM.render(<MissedCard
                    title="Do Something You Can't Win At"
                    description="And make it something you love"
                    image="https://cdn-images-1.medium.com/max/800/1*cioT9CBBV2s0svet9LqKZA.png"
                    link="https://medium.com/@krisgage/do-something-you-cant-win-at-cb2e5d2ad3c1"
                    author={author}
                    postedDate="2018-03-30T07:00:00.000Z"
                    minutesToRead={minutes}
                    hasAudioAvailable={boolean}
                    memberPreview={boolean} />, div); */
        ReactDOM.render(element, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('should render with title, description, image, etc.', () => {
        const element = testCard.map((card, index) => {
            return <MissedCard key={index} missedcard={card} />
        });
        /*         const minutes = 3;
                const boolean = false;
                const author = {
                    name: "Kris Gage",
                    image: "https://cdn-images-1.medium.com/fit/c/80/80/1*e7o4kbk8ofT6eKaLrO0jaw.png",
                    isMediumMember: false
                };
        
                const component = TestRenderer.create(<MissedCard
                    title="Do Something You Can't Win At"
                    description="And make it something you love"
                    image="https://cdn-images-1.medium.com/max/800/1*cioT9CBBV2s0svet9LqKZA.png"
                    link="https://medium.com/@krisgage/do-something-you-cant-win-at-cb2e5d2ad3c1"
                    author={author}
                    postedDate="2018-03-30T07:00:00.000Z"
                    minutesToRead={minutes}
                    hasAudioAvailable={boolean}
                    memberPreview={boolean} />); */
        const component = TestRenderer.create(element);
        expect(component.toJSON()).toMatchSnapshot();
    });
});    
