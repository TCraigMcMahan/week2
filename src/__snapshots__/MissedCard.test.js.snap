// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissedCard should render with title, description, image, etc. 1`] = `
<div
  className="Missed-row"
>
  <div
    className="missedUpperScreen"
  >
    <img
      alt=""
      className="Missed-Card-img"
      src="https://cdn-images-1.medium.com/max/800/1*cioT9CBBV2s0svet9LqKZA.png"
    />
    <div
      className="missedTitleStyle"
    >
       
      Do Something You Can't Win At
       
    </div>
    <br />
    <div
      className="missedDescriptionStyle"
    >
      And make it something you love
    </div>
  </div>
  <div
    className="missedAuthorInfo"
  >
    <div>
      <img
        alt=""
        className="Author-img"
        src="https://cdn-images-1.medium.com/fit/c/80/80/1*e7o4kbk8ofT6eKaLrO0jaw.png"
      />
    </div>
    <div
      className="authorInfo-namedate"
    >
      <div
        className="authorNameStyle"
      >
        Kris Gage
      </div>
      <div
        align="left"
      >
         
        Mar 30
         - 
        3
         min read
      </div>
    </div>
    <div
      className="bookmarkContainer"
    >
      <img
        alt="Bookmark"
        className="bookmarkStyle"
        src="icon-bookmark.svg"
      />
    </div>
  </div>
</div>
`;
